### DISCUSSION TITLE
Conceptual Doubt regarding Catastrophic forgetting

### DISCUSSION THREAD

**Discussion by Parth-tech** [explosion/spaCy#8515] [Help: Coding & Implementations] [answered]
I just encountered this in the discussion and I need little bit of clarification on it.

Suppose I train a model for NER and I have 5 entities consider **_A, B. C, D & E_** 

So post that I if want to retrain it then the dataset that I use for retaining has **_A, B. C, D & E_** and few others.

So the string corresponding to these entities ( A ..... E ) must retained or any string pointing to these label would do fine?

Thank You!

**Comment by svlandeg** [comment, marked as answer]
I'm not sure I understand the question.

Catastrophic forgetting is what happens when the machine learning model forgets examples it was originally trained on. This occurs when you retrain the model on new data, and you don't feed it some of those original (or similar) examples. For instance, if you'd have an NER trained on A, B, C, D and E, but you continue training on only A and B, then eventually it will "unlearn" how to predict labels C, D and E, because it will focus on getting A and B correctly. This isn't really about the string representations of the labels, but more about the type of examples that are being fed into the model, and how the model adjusts its internal weights to learn on the new samples.

**Reply by Parth-tech** [reply]
Got it @svlandeg! Thanks for the explanation it cleared my doubt!
